pkg/plugin/aggregation/aggregator.go

```diff
@@ -261,7 +261,7 @@ func (a *Aggregator) IngestResults(ctx context.Context, resultsCh <-chan *plugin
 		case result, more = <-resultsCh:
 		}
 
-		if !more {
+		if result == nil && !more {
```

Comment:
Should always be the same as before, but wasn't sure if there was ever, somehow, that you'd get the `result, !more` instead of `nil, !more`. Just an extra clear check that we will process each result.